os/net/lwip/src/api/sockets.c

```diff
@@ -453,28 +468,31 @@ struct lwip_sock *trycheck_selwait_socket(int s)
  */
 int alloc_socket(struct netconn *newconn, int accepted)
 {
-	int i;
+	int idx;
+	struct socketlist *list;
+	struct lwip_sock *sock;
+
+	list = sched_getsockets();
 	SYS_ARCH_DECL_PROTECT(lev);
+	sock = (struct lwip_sock *)kmm_zalloc(sizeof(struct lwip_sock));
+
+	if (!sock) {
+		ndbg("memory is not sufficient for sockets\n");
+		return -1;
+	}
 
-	/* allocate a new socket identifier */
-	for (i = 0; i < NUM_SOCKETS; ++i) {
-		/* Protect socket array */
+	if (list) {
```

Comment:
Don't you need to add else case to free the sock for error case?

Reply:
thanks and corrected.